### PR TITLE
feat: Dashboard page html + state

### LINKED ISSUE #14: Dashboard page
## Description
- [x] Html
- [ ] Styles
- [x] State handling
- [x] Data fetching
- [x] Pop-up feature

### PR DESCRIPTION
**Pull request by jcmtallon**
## Description
- [x] Dashboard page html (no styles yet)
- [x] State handling (initial draft)
- [x] Data fetching (using mock data first)
- [x] Remove unused create-next-app template elements

## Related Issue
Resolves #14

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -1 +1,2 @@
-API_BASE_URL=http://localhost:23500
+NEXT_PUBLIC_API_BASE_URL=http://localhost:23500
+NEXT_PUBLIC_FETCH_MOCK_DATA=true
```
Comment (jcmtallon) [inline .env.development:2]: Now that I come to think about it, `NEXT_PUBLIC_FETCH_MOCK_DATA`  should not be updated to the repository, as this is a value that each developer should be free to switch on and off during development. Will move it to some local environment file in the following PR.

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import React from 'react';
+
+export const Footer = () => <footer>Footer text</footer>;
```
Comment (jcmtallon) [inline src/components/Footer.tsx:3]: Whatever message/copyright you would like to display in the footer, feel free to let me know anytime you want. Footer can vanish if you feel there is no need for it.

File: src/components/Head.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+interface OwnProps {
+  title?: string;
+  description?: string;
+}
+
+export const Head = ({ title, description }: OwnProps) => (
+  <NextHead>
+    <title>{title || 'Power Guard Monitor'}</title>
+    <meta name="description" content={description || 'Power Guard Plant Monitoring Site'} />
+    <link rel="icon" href="/favicon.ico" />
```
Comment (jcmtallon) [inline src/components/Head.tsx:13]: Please let me know if you want to display a different title or description.

Also, took the liberty to make a favicon with the same image as this repository. 
image

File: src/components/pages/dashboard/DashboardTopArea.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <div>Data Collection Process</div>
+            <div>{done ? 'Complete' : 'Running'}</div>
+          </div>
+          <div>
+            <div>Started</div>
+            <div>{proc_start !== null ? proc_start : '-'}</div>
+          </div>
+          <div>
+            <div>Finished</div>
+            <div>{proc_end !== null ? proc_end : '-'}</div>
```
Comment (jcmtallon) [inline src/components/pages/dashboard/DashboardTopArea.tsx:28]: Simply displaying raw data right now. I will add format, style, validation, etc. in future pull requests.

File: src/services/apiCore.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import axios from 'axios';
 
 export const axiosInstance = axios.create({
-  baseURL: process.env.API_BASE_URL,
+  baseURL: process.env.NEXT_PUBLIC_API_BASE_URL,
```
Comment (jcmtallon) [inline src/services/apiCore.ts:4]: Environment variables in nextjs must start with `NEXT_PUBLIC_` to be accessible on the browser side.

**Review by jcmtallon — COMMENTED** [review]
Added some comments to the code to provide you with more context.

**Comment by jcmtallon** [comment]
Merging this pull request so I can continue working on top of it 
You can always go to Pull Requests > Closed in case you want to give it a look later.

File: src/components/Footer.tsx (hunk 2 above)
Comment (power-guard) [reply src/components/Footer.tsx:3]: Feels like a good place for attribution to the front-end designer.

File: src/components/Head.tsx (hunk 3 above)
Comment (power-guard) [reply src/components/Head.tsx:13]: Everything looks good.

**Pull request merged**